src/__tests__/core/forgotPassword.test.ts

```
@@ -0,0 +1,432 @@
+      await expect(resetPasswordWithCode(
+        'test@example.com',
+        resetCode,
+        'SamePassword123!',
+        { preventReuse: true }
+      )).rejects.toThrow('New password cannot be the same as your current password');
+    });
+
+    it('should handle database adapter without updateUser method', async () => {
+      // Save original method and remove it
+      const originalUpdateUser = mockAdapter.updateUser;
+      delete (mockAdapter as any).updateUser;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)